src/AzureClient/AzureClient.cs

```diff
@@ -1,24 +1,249 @@
+        {
+            ConnectionString = string.Empty;
+            ActiveTargetName = string.Empty;
+        }
+
+        /// <inheritdoc/>
+        public async Task<ExecutionResult> ConnectAsync(
+            IChannel channel,
+            string subscriptionId,
+            string resourceGroupName,
+            string workspaceName,
+            string storageAccountConnectionString,
+            bool forceLogin = false)
+        {
+            ConnectionString = storageAccountConnectionString;
+
+            var clientId = "84ba0947-6c53-4dd2-9ca9-b3694761521b"; // Microsoft Quantum Development Kit
+            var authority = "https://login.microsoftonline.com/common";
+            var msalApp = PublicClientApplicationBuilder.Create(clientId).WithAuthority(authority).Build();
+
+            // Register the token cache for serialization
+            var cacheFileName = "aad.bin";
+            var cacheDirectoryEnvVarName = "AZURE_QUANTUM_TOKEN_CACHE";
+            var cacheDirectory = System.Environment.GetEnvironmentVariable(cacheDirectoryEnvVarName);
+            if (string.IsNullOrEmpty(cacheDirectory))
+            {
+                cacheDirectory = Path.Join(System.Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile), ".azure-quantum");
+            }
+
+            var storageCreationProperties = new StorageCreationPropertiesBuilder(cacheFileName, cacheDirectory, clientId).Build();
+            var cacheHelper = await MsalCacheHelper.CreateAsync(storageCreationProperties);
+            cacheHelper.RegisterCache(msalApp.UserTokenCache);
+
+            var scopes = new List<string>() { "https://quantum.microsoft.com/Jobs.ReadWrite" };
+
+            bool promptForLogin = forceLogin;
+            if (!promptForLogin)
+            { 
+                try
+                {
+                    var accounts = await msalApp.GetAccountsAsync();
+                    AuthenticationResult = await msalApp.AcquireTokenSilent(
+                        scopes, accounts.FirstOrDefault()).WithAuthority(msalApp.Authority).ExecuteAsync();
+                }
+                catch (MsalUiRequiredException)
+                {
+                    promptForLogin = true;
+                }
+            }
+
+            if (promptForLogin)
+            {
+                AuthenticationResult = await msalApp.AcquireTokenWithDeviceCode(scopes,
+                    deviceCodeResult =>
+                    {
+                        channel.Stdout(deviceCodeResult.Message);
+                        return Task.FromResult(0);
+                    }).WithAuthority(msalApp.Authority).ExecuteAsync();
+            }
+
+            if (AuthenticationResult == null)
+            {
+                return AzureClientError.AuthenticationFailed.ToExecutionResult();
+            }
+
+            var credentials = new Rest.TokenCredentials(AuthenticationResult.AccessToken);
+
+            QuantumClient = new QuantumClient(credentials);
```

Comment:
```suggestion
            QuantumClient = new QuantumClient(credentials)
            {
                SubscriptionId = subscriptionId,
                ResourceGroupName = resourceGroupName,
                WorkspaceName = workspaceName
            };
```
In general, I suggest using initialization syntax following constructors to help prepare for [C# 9 init-only properties](https://github.com/dotnet/csharplang/blob/master/proposals/init.md).

Reply:
Thanks! I applied your suggestion, but CodeFlow wasn't showing me that this was actually a suggestion, so I didn't use the commit-suggestion feature. I'll give you credit through a mention here: @cgranade   :-)